fideslog/api/models/models.py

```diff
@@ -53,6 +53,24 @@ class AnalyticsEvent(Base):
     )
 
 
+class UserRegistrationEvent(Base):
+    """
+    The persisted details about a user registration event.
+    """
+
+    __tablename__ = "USER_REGISTRATION_EVENTS"
+
+    id = Column(
+        "USER_REGISTRATION_ID", Integer, Sequence("event_id_seq"), primary_key=True
+    )
+    analytics_id = Column("ANALYTICS_ID", String, default=None, nullable=True)
+    email = Column("EMAIL", String, default=None, nullable=True)
+    organization = Column("ORGANIZATION", String, default=None, nullable=True)
```

Comment:
> we can instead just implement a much more robust/scalable telemetry service, like OpenTelemetry, and save ourselves lots of time/money/effort

Just to clarify here - there's nothing inherently anti-privacy about OpenTelemetry. As I stated in my original fideslog design doc, everything being done by fideslog can be done by a system like OpenTelemetry + a timeseries DB + Grafana, or similar. We should absolutely think about deprecating fideslog in favor of a distributed telemetry system when the request burden reaches a demand that we don't want to worry about supporting.

Reply:
These are all very reasonable privacy concerns, and you've just not been involved in the run up to this. Here's the quick background...

We've decided to collect opt-in consent to collect identifiable `user.contact` data from users when they first elect to engage with Fides, so that we can contact them for both product feedback (ie `improve.system`) and direct marketing (ie `advertising.first_party`). There is nothing inherently wrong with either of those as long as we collect their consent, allow them to revoke it at any time, and protect that data accordingly. Additionally, I think `fideslog` should continue to act as a reference example for how to do this in a pragmatic, privacy-respecting way.

I was very specific about this design, though, and want to ensure that we do store the analytics events and user registration rows in distinct tables, so we only ever have a single copy of the user data linked to the `analytics_id`. The combination of `user.contact` data and the pseudonymous `user.unique_id` does mean that we've linked an otherwise pseudonymous analytics event to an identifiable user (which is precisely the new `data_use` we are enabling). However, by storing these singular user records in a separate registration table like this, we give ourselves the ability to delete a single row and fully pseudonymize that analytics history again, since the link has been lost. That allows us to easily process erasure requests against this system without much effort if a user does elect to opt-out after first providing their consent.

Alternatively, we could collect and store the user registration data in a separate system (eg a marketing platform) but I'd prefer to avoid spreading it around and keep this fully transparent. After all we're trying to showcase privacy engineering best practices here 👍

To your other points though:
* Agreed that encrypting these rows at rest is important. Can we update this to use standard SQLAlchemy helpers for this and supply a secret key to the deployment? Several examples of this in the core `fides` repo
* Deprecating this to move to an alternative option next year sounds reasonable to me, if we feel it's not technically feasible for this service to process any identifiable data in a safe way. It's not our core objective of course, so we can work on other problems instead. That said, I do feel we should be up to the challenge of storing these two types of records in a privacy-respecting way, but I'm open to a roadmap that moves away from here next year 
* The purpose of this data processing will absolutely be communicated to the user upon registration when we prompt for their consent; you've seen where this is done in the CLI today, and we'll be amending the language to clarify the updated data uses. But to reiterate again: it's for product feedback and marketing use. There's nothing cloak-and-dagger here, it's a necessary part of our strategy to grow the OSS market
* Performance of this has been acceptable thus far even after collecting millions of events. We can do some client-side filtering, honestly, since I'm noticing that we're definitely recording too many healthcheck API calls!